lib/opam_solve.ml

```diff
@@ -175,6 +178,22 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
     in
     regular_versions @ avoid_versions
 
+  let prefer_version ~version versions =
```

Comment:
`preferred_version` would be nicer here. Maybe even something like `let promote_version ~preferred_version candidate_versions =` as a analogue to `demote`.

Reply:
`preferred_version` would be confusing here as it just re-orders the list and that name would somehow indicate it will return a version.

I initially thought about naming it `promote_version` so let's go for this one which is indeed better than the current name!

Reply:
Oh, yeah, I meand `preferred_version` as the name of the named argument, not the name of the function. I think "promote" captures what the function is attempting to do pretty well!

Reply:
I feel like if we use `promote_version` than naming the argument `~preferred_version` is a bit redundant.